pkg/operator/datasources.go

```diff
@@ -440,6 +447,67 @@ func (op *Reporting) handlePrestoTableDataSource(logger log.FieldLogger, dataSou
 	return nil
 }
 
+// handleLinkExistingTable is reponsible for managing a linkExistingTable ReportDataSource sub-type.
+// When a new custom resource is detected, we first validate the @dataSource object and check if the
+// tableName is in the form of a fully-qualified table name. In the case where the operator hasn't
+// processed this resource before, query the Presto table's metadata and then create an unmanaged
+// PrestoTable custom resource with the columns returned from the query. Once the PrestoTable resource
+// has been created, update the @dataSource Status field to refer to the name of the created PrestoTable.
+func (op *Reporting) handleLinkExistingTable(logger log.FieldLogger, dataSource *metering.ReportDataSource) error {
+	if dataSource.Spec.LinkExistingTable.TableName == "" {
+		return fmt.Errorf("invalid configuration passed: spec.linkExistingTable.tableName field cannot be empty")
+	}
+	inputs := strings.Split(dataSource.Spec.LinkExistingTable.TableName, ".")
+	if len(inputs) != expectedArrSplitElementsFQTN {
+		return fmt.Errorf("invalid configuration passed: spec.linkExistingTable.tableName is not a fully-qualified table name")
+	}
+
+	// check if this resource has already been processed and we can exit early
+	if dataSource.Status.TableRef.Name != "" {
+		prestoTable, err := op.prestoTableLister.PrestoTables(dataSource.Namespace).Get(dataSource.Status.TableRef.Name)
+		if err != nil {
+			return fmt.Errorf("failed to get the %s PrestoTable listed in the %s ReportDataSource Status: %v", dataSource.Status.TableRef.Name, dataSource.Name, err)
+		}
+		tableName, err := reportingutil.FullyQualifiedTableName(prestoTable)
+		if err != nil {
+			return err
+		}
+		logger.Infof("existing LinkExistingTable ReportDataSource discovered, tableName: %s", tableName)
+		return nil
+	}
+
+	catalog := inputs[0]
+	schema := inputs[1]
+	tableName := inputs[2]
+
+	// using the fully-qualified table name from the resource, verify we can query the existing table's
+	// properties for its metadata. We can then use that information and create a PrestoTable resource.
+	cols, err := op.prestoTableManager.QueryMetadata(catalog, schema, tableName)
+	if err != nil {
+		return fmt.Errorf("failed to query the %s.%s.%s Presto table metadata: %v", catalog, schema, tableName, err)
+	}
+	prestoTable, err := op.createPrestoTableCR(dataSource, metering.ReportDataSourceGVK, catalog, schema, tableName, cols, true, false, "")
```

Comment:
The first `true` is actually fairly important to note, it's setting `unmanaged` bool. Perhaps worth a comment or constant. Had to look up the func definition to know that.

Reply:
Done - added both a comment and some variables:
```go
	var createView bool
	catalog := inputs[0]
	schema := inputs[1]
	tableName := inputs[2]
	unmanagedTable := true

	// using the fully-qualified table name from the resource, verify we can query the existing table's
	// properties for its metadata. We can then use that information and create a PrestoTable resource.
	cols, err := op.prestoTableManager.QueryMetadata(catalog, schema, tableName)
	if err != nil {
		return fmt.Errorf("failed to query the %s.%s.%s Presto table metadata: %v", catalog, schema, tableName, err)
	}
	// attempt to create an unmanaged PrestoTable CR as we're linking an existing table in Presto to
	// this particular ReportDataSource and don't need the reporting-operator to create this table for us.
	prestoTable, err := op.createPrestoTableCR(dataSource, metering.ReportDataSourceGVK, catalog, schema, tableName, cols, unmanagedTable, createView, "")
	if err != nil {
		return fmt.Errorf("failed to create the PrestoTable for the %s ReportDataSource: %v", dataSource.Name, err)
	}
```